scipy/ndimage/_rank_filter_1d.pyx

```diff
@@ -0,0 +1,31 @@
+cimport cython
+
+ctypedef fused numeric_t:
+    cython.float
+    cython.double
+    cython.int
+    cython.uint
+    cython.ulong
+    cython.long
+    cython.longlong
+    cython.ulonglong
+    cython.char
+    cython.uchar
+    cython.short
+    cython.ushort
+    cython.longdouble
+
+cdef extern from "rank_filter_1d.cpp" nogil:
+    void rank_filter[T](T* in_arr, int rank, int arr_len, int win_len, T* out_arr, int mode, T cval, int origin)
+
+def rank_filter_1d_cpp_api(numeric_t[:] in_arr, int rank, int win_len, numeric_t[:] out_arr, int mode, numeric_t cval=0, int origin=0):
+    rank_filter(&in_arr[0], rank, in_arr.shape[0], win_len, &out_arr[0], mode, cval, origin)
+    return out_arr
+
+def rank_filter_1d(x: np.array, rank, footprint: np.array, x_out: np.array, mode: int, cval=0.0, origin=0):
```

Comment:
type is `np.ndarray` rather than `np.array` (this showed up as a build warning)

Reply:
fixed